core/src/main/java/feast/core/controller/CoreServiceRestController.java

```diff
@@ -0,0 +1,199 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.controller;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Timestamp;
+import feast.core.config.FeastProperties;
+import feast.core.model.Project;
+import feast.core.service.AccessManagementService;
+import feast.core.service.JobService;
+import feast.core.service.SpecService;
+import feast.core.service.StatsService;
+import feast.proto.core.CoreServiceProto.GetFeastCoreVersionResponse;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsRequest.Builder;
+import feast.proto.core.CoreServiceProto.GetFeatureStatisticsResponse;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsRequest;
+import feast.proto.core.CoreServiceProto.ListFeatureSetsResponse;
+import feast.proto.core.CoreServiceProto.ListFeaturesRequest;
+import feast.proto.core.CoreServiceProto.ListFeaturesResponse;
+import feast.proto.core.CoreServiceProto.ListProjectsResponse;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneOffset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * EXPERIMENTAL: Controller for HTTP endpoints to Feast Core. These endpoints are subject to change.
+ */
+@RestController
+@CrossOrigin
+@Slf4j
+@RequestMapping(value = "/api/v1", produces = "application/json")
+public class CoreServiceRestController {
+
+  private final FeastProperties feastProperties;
+  private SpecService specService;
+  private JobService jobService;
+  private StatsService statsService;
+  private AccessManagementService accessManagementService;
+
+  @Autowired
+  public CoreServiceRestController(
+      FeastProperties feastProperties,
+      SpecService specService,
+      JobService jobService,
+      StatsService statsService,
+      AccessManagementService accessManagementService) {
+    this.feastProperties = feastProperties;
+    this.specService = specService;
+    this.jobService = jobService;
+    this.statsService = statsService;
+    this.accessManagementService = accessManagementService;
+  }
+
+  /**
+   * GET /version : Fetches the version of Feast Core.
```

Comment:
Is there a reason to write out the path and verb in the comments? Seems like extra documentation to maintain when its already defined in code.

Reply:
If the javadoc will eventually be user-facing, including the path and verb in the javadoc would be useful for end-users. However if it is to annotate the code then yep it could feel redundant.

Reply:
there are definitely tools to generate the openapi yaml and use that for documentation rather than depending upon comments for the routes, response codes, etc...

Reply:
It is definitely good to have. Opened an issue (https://github.com/feast-dev/feast/issues/939) to track this.